File: Python/Problem4.py
class Solution:
    def shortestPalindrome(self, s: str) -> str:
        if s[::-1]==s:
            return s
        def is_palindrome(s,r):
            if s[:r+1]== s[:r+1][::-1]:
                return True
            return False
        for r in range(len(s)-1,-1,-1):
            if is_palindrome(s,r):
                suffix= s[r+1:]
                return suffix[::-1]+s
            
     

File: Java/Problem4.java
class Solution {
    public String shortestPalindrome(String s) {
        if (new StringBuilder(s).reverse().toString().equals(s)){
            return s;
        }
        for (int r=s.length()-1; r>=0;r--){
            if (isPalindrome(s,r)){
                String suffix = s.substring(r+1);
                return new StringBuilder(suffix).reverse().toString()+s;
            }
        }
        return "";
    }
    private Boolean isPalindrome(String s, int r) {
        String prefix = s.substring(0,r+1);
        return new StringBuilder(prefix).reverse().toString().equals(prefix);
    }
}